src/setup/building/test/building_test.go

```diff
@@ -12,7 +14,9 @@ func TestBuildFunctionJava(t *testing.T) {
 
 func TestBuildFunctionGolang(t *testing.T) {
 	b := &building.Builder{}
-	b.BuildFunction("test/function/path", "golang")
+	b.BuildFunction("resources/hellogo", "go1.x")
```

Comment:
For BuildFunction, we can use the same source code dir:

```golang
func (b *Builder) BuildFunction(provider string, functionName string, runtime string) {
	functionPath := fmt.Sprintf("setup/deployment/raw-code/serverless/%s/%s", provider, functionName)
        // ...
        buildGolang(functionPath)
```

For testing, we want to working dir to be at `src` instead of the test folder. Check [this](https://stackoverflow.com/questions/23847003/golang-tests-and-working-directory) post for some ideas on how to achieve this.

Afterwards,

```golang
func TestBuildFunctionGolang(t *testing.T) {
	b := &building.Builder{}
	b.BuildFunction("aws", "hellogo", "go1.x")
```

Reply:
Thanks for the suggestion, I changed the parameters for `BuildFunction`, updated the tests and deleted the extra hellogo function that was in the test resources directory